src/components/dropdown/dropdown.svelte

```diff
@@ -129,58 +131,61 @@
 </script>
 
 <div class="leo-dropdown">
-  <FormItem
-    bind:disabled
-    bind:required
-    bind:size
-    {mode}
-    showFocusOutline={isOpen}
-  >
-    <slot name="label" slot="label" />
-    <slot name="left-icon" slot="left-icon" />
-    <button
-      bind:this={button}
-      class="click-target"
-      {disabled}
-      on:click|stopPropagation={(e) => (isOpen = !isOpen)}
+  <div bind:this={dropdown}>
+    <FormItem
+      bind:disabled
+      bind:required
+      bind:size
+      {mode}
+      showFocusOutline={isOpen}
     >
-      {#if value !== undefined}
-        <slot name="value" {value}>
-          <span class="value">{valueText}</span>
-        </slot>
-      {:else}
-        <slot name="placeholder">
-          <span class="placeholder">{placeholder}</span>
-        </slot>
-      {/if}
-    </button>
-    <slot name="right-icon" slot="right-icon">
-      <div class="indicator" class:open={isOpen}>
-        <Icon name="arrow-small-down" />
-      </div>
-    </slot>
-    <div class="menu" slot="after">
-      <div
-        class="leo-dropdown-popup"
-        hidden={!isOpen}
-        transition:scale={{ duration: 60, start: 0.8 }}
-        use:clickOutside={isOpen &&
-          ((e) => {
-            e.stopPropagation()
-            e.preventDefault()
-            isOpen = false
-          })}
-        on:keypress={(e) => {
-          if (e.code !== 'Enter' && e.code !== 'Space') return
-          selectOption(e)
-        }}
-        on:click|stopPropagation|preventDefault={selectOption}
-        bind:this={popup}
+      <slot name="label" slot="label" />
+      <slot name="left-icon" slot="left-icon" />
+      <button
+        bind:this={button}
+        class="click-target"
+        {disabled}
+        on:click|stopPropagation={(e) => (isOpen = !isOpen)}
       >
-        <slot />
-      </div>
+        {#if value !== undefined}
+          <slot name="value" {value}>
+            <span class="value">{valueText}</span>
+          </slot>
+        {:else}
+          <slot name="placeholder">
+            <span class="placeholder">{placeholder}</span>
+          </slot>
+        {/if}
+      </button>
+      <slot name="right-icon" slot="right-icon">
+        <div class="indicator" class:open={isOpen}>
+          <Icon name="arrow-small-down" />
+        </div>
+      </slot>
+    </FormItem>
+  </div>
+  <Floating target={dropdown} placement="bottom-start">
+    <div
+      class="leo-dropdown-popup"
+      hidden={!isOpen}
+      style:min-width="{dropdown?.getBoundingClientRect().width}px"
```

Comment:
This isn't getting updated when dropdown width changes. Make storybook window narrower, then click it open and popup will be too wide.

Reply:
<img width="637" alt="image" src="https://github.com/brave/leo/assets/8665427/231cea1e-3678-4ed5-a5aa-33f91a69931a">

Also, the content overflows. Perhaps, add shadow, rounding and border in the outerbox?

Reply:
@nullhook I'm tempted to leave that as a follow up, because I'm going on PTO next week, and it's like that even without this PR - we should definitely fix it though